buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -59,22 +102,282 @@ public Distribution getDistribution() {
+            throw new TestClustersException("Can not start " + this + ", missing: " + distroArtifact);
+        }
+        if (distroArtifact.isDirectory() == false) {
+            throw new TestClustersException("Can not start " + this + ", is not a directory: " + distroArtifact);
+        }
+        services.sync(spec -> {
+            spec.from(new File(distroArtifact, "config"));
+            spec.into(getConfigFile().getParent());
+        });
+        writeConfigFile();
+        startElasticsearchProcess(distroArtifact);
+        registerCleanupHooks();
     }
 
-    void stop(boolean tailLogs) {
+    private void startElasticsearchProcess(File distroArtifact) {
+        logger.info("Running `bin/elasticsearch` in `{}` for {}", workingDir, this);
+        final ProcessBuilder processBuilder = new ProcessBuilder();
+        if (OperatingSystem.current().isWindows()) {
+            processBuilder.command(
+                "cmd", "/c",
+                new File(distroArtifact, "\\bin\\elasticsearch.bat").getAbsolutePath()
+            );
+        } else {
+            processBuilder.command(
+                new File(distroArtifact.getAbsolutePath(), "bin/elasticsearch").getAbsolutePath()
+            );
+        }
+        try {
+            processBuilder.directory(workingDir);
+            Map<String, String> environment = processBuilder.environment();
+            // Don't inherit anything from the environment for as that would  lack reproductability
+            environment.clear();
+            if (javaHome != null) {
+                environment.put("JAVA_HOME", getJavaHome().getAbsolutePath());
+            } else if (System.getenv().get("JAVA_HOME") != null) {
+                logger.warn("{}: No java home configured will use it from environment: {}",
+                    this, System.getenv().get("JAVA_HOME")
+                );
+                environment.put("JAVA_HOME", System.getenv().get("JAVA_HOME"));
+            } else {
+                logger.warn("{}: No javaHome configured, will rely on default java detection", this);
+            }
+            environment.put("ES_PATH_CONF", getConfigFile().getParentFile().getAbsolutePath());
+            environment.put("ES_JAVA_OPTIONS", "-Xms512m -Xmx512m");
+            // don't buffer all in memory, make sure we don't block on the default pipes
+            processBuilder.redirectError(ProcessBuilder.Redirect.appendTo(getStdErrFile()));
+            processBuilder.redirectOutput(ProcessBuilder.Redirect.appendTo(getStdoutFile()));
+            esProcess = processBuilder.start();
+        } catch (IOException e) {
+            throw new TestClustersException("Failed to start ES process for " + this, e);
+        }
+    }
+
+    private void registerCleanupHooks() {
+        requireNonNull(esProcess, "Attempted to register cleanup hooks without a process");
+        // The Gradle daemon interrupts all threads from the build when a build concludes,
```

Comment:
I just tried this locally, the daemon doesn't seem to interrupt this kind of `Thread` if the build concludes successfully. So we'd be leaking a `Thread` here for each ES node we spin up when using the daemon?
Maybe a better Pattern would be to just use a single `Thread` for this globally, keep a static reference to it and a collection that holds all the running Es nodes and then have that thread stop all nodes in that collection on interrupt?

Reply:
This is now fixed, thanks!